pkg/build/api/validation/validation.go

```diff
@@ -51,6 +51,17 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) errs.ValidationErrorList
 	return allErrs
 }
 
+func ValidateBuildRequest(request *buildapi.BuildRequest) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if len(request.BuildConfigName) == 0 && len(request.BuildName) == 0 {
```

Comment:
yeah i missed the use case here where you're creating a new build from an existing build.  make sense.